src/flint/fmpz_mod_mat.jl

```diff
@@ -280,6 +288,20 @@ function mul!(z::Vector{ZZRingElem}, a::Vector{ZZRingElem}, b::T) where T <: Zmo
    return z
 end
 
+function Generic.add_one!(a::ZZModMatrix, i::Int, j::Int)
+  @boundscheck Generic._checkbounds(a, i, j)
+  GC.@preserve a begin
+    x = mat_entry_ptr(a, i, j)
+    ccall((:fmpz_mod_add_si, libflint), Nothing,
+          (Ptr{ZZModRingElem}, Ptr{ZZModRingElem}, Int, Ref{ZZModRing}),
+          x, x, 1, base_ring(a))
+    ccall((:fmpz_mod, libflint), Nothing,
```

Comment:
Are you sure this is necessary? Looking at https://github.com/flintlib/flint/blob/flint-2.9/fmpz_mod/add_fmpz.c#L42, it seems that `fmpz_mod_add_si` does a reduction.

Reply:
I was actually wondering about this:
```
julia> R, _ = residue_ring(ZZ, ZZ(3));

julia> a = zero_matrix(R, 2, 2);

julia> a[1, 1] = 2;

julia> i = 1; j = 1;

julia> GC.@preserve a begin
           x = Nemo.mat_entry_ptr(a, i, j)
           ccall((:fmpz_mod_add_si, Nemo.libflint), Nothing,
                 (Ptr{ZZModRingElem}, Ptr{ZZModRingElem}, Int, Ref{ZZModRing}),
                 x, x, 1, base_ring(a))
       end

julia> a
[3   0]
[0   0]

```

Reply:
I don't understand this function.
```
julia> R, _ = residue_ring(ZZ, ZZ(3));

julia> x = R(); y = R();

julia> ccall((:fmpz_mod_add_si, Nemo.libflint), Nothing,
                        (Ref{ZZModRingElem}, Ref{ZZModRingElem}, Int, Ref{ZZModRing}),
                        y, x, 1, R)

julia> y
0
```